src/components/organisms/PageHeader/PageHeader.tsx

```diff
@@ -113,30 +113,34 @@ const PageHeader = () => {
     shell.openExternal(url);
   }, [autosavingError]);
 
-  const initGitRepo = async () => {
+  const initGitRepoHandler = async () => {
     if (!projectRootFolder) {
       return;
     }
 
     trackEvent('git/initialize');
     setIsInitializingGitRepo(true);
 
-    const result = await promiseFromIpcRenderer('git.initGitRepo', 'git.initGitRepo.result', projectRootFolder);
-
-    if (result.error) {
-      showGitErrorModal('Failed to initialize git repo');
+    try {
+      await initGitRepo({path: projectRootFolder});
+    } catch (e: any) {
+      showGitErrorModal('Failed to initialize git repo', e.message);
       setIsInitializingGitRepo(false);
       return;
     }
 
     monitorGitFolder(projectRootFolder, store);
 
-    promiseFromIpcRenderer('git.getGitRepoInfo', 'git.getGitRepoInfo.result', projectRootFolder).then(repo => {
-      dispatch(setRepo(repo));
-      dispatch(setCurrentBranch(repo.currentBranch));
-      setIsInitializingGitRepo(false);
-      dispatch(updateProjectsGitRepo([{path: projectRootFolder, isGitRepo: true}]));
-    });
+    try {
+      getRepoInfo({path: projectRootFolder || ''}).then(repo => {
```

Comment:
I don't think this try/catch will work if you're not using `await`
if you're using `.then` then you have to use `.catch()`